flopy/mf6/utils/codegen/filters.py

```diff
@@ -34,528 +34,526 @@ def enter(p, k, v):
-            elif _type == "record":
-                children = ", ".join(
-                    [v["name"] for v in children.values()]
-                )
-                return f"({children})"
-            elif _type == "keystring":
-                return " | ".join([v["name"] for v in children.values()])
-        elif shape:
-            return f"[{_type}]"
-        return var["type"]
-
-    def children(var: dict) -> Optional[dict]:
-        _type = var["type"]
-        items = var.get("items", None)
-        fields = var.get("fields", None)
-        choices = var.get("choices", None)
-        if items:
-            assert _type == "recarray"
-            return items
-        if fields:
-            assert _type == "record"
-            return fields
-        if choices:
-            assert _type == "keystring"
-            return choices
-        return None
-
-    def default_value(var: dict) -> Any:
-        _default = var.get("default", None)
-        if _default is not None:
-            return _default
-        return None
+    # if tagged, remove the leading keyword
+    elif tagged:
+        keyword = next(iter(fields), None)
+        if keyword:
+            fields.pop(keyword)
+
+    # if the record represents a file...
+    elif "file" in name:
+        # remove filein/fileout
+        field_names = list(fields.keys())
+        for term in ["filein", "fileout"]:
+            if term in field_names:
+                fields.pop(term)
+
+        # remove leading keyword
+        keyword = next(iter(fields), None)
+        if keyword:
+            fields.pop(keyword)
+
+    var["fields"] = fields
+    return var
+
+def type(var: dict) -> str:
+    """
+    Get a readable representation of the variable's type.
+    TODO: eventually replace this with a proper `type` in
+    the variable spec when we add type hints
+    """
+    _type = var["type"]
+    shape = var.get("shape", None)
+    children_vars = children(var)
+    if children_vars:
+        if _type == "recarray":
+            if len(children_vars) == 1:
+                first = list(children_vars.values())[0]
+                if first["type"] in ["record", "keystring"]:
+                    return f"[{type(first)}]"
+            children_vars = ", ".join(
+                [v["name"] for v in children_vars.values()]
+            )
+            return f"[{children_vars}]"
+        elif _type == "record":
+            children_vars = ", ".join(
+                [v["name"] for v in children_vars.values()]
+            )
+            return f"({children_vars})"
+        elif _type == "keystring":
+            return " | ".join([v["name"] for v in children_vars.values()])
+    elif shape:
+        return f"[{_type}]"
+    return var["type"]
+
+def children(var: dict) -> Optional[dict]:
+    _type = var["type"]
+    assert "items" not in var, 'TODO: fix bug to use "item" (not "items")'
+    items = var.get("items", None)  # TODO: item = var.get("item", None)
```

Comment:
I'll remove the `assert` statement to not break from upstream changes to modflow-devtools.